### PR TITLE
TextField icon click gives focus to input

### PR DESCRIPTION
**Pull request by mattpauldavies**
**Backwards Compatibility Implications**

_None_

**New Features**

_None_

**Bug Fixes**

- TextField icon click gives input focus

### CODE REVIEW

File: src/components/Fields/TextField.js [hunk 1]
```diff
@@ -265,19 +278,24 @@ class TextField extends Component {
         const icon = this.props.icon ?
             (
                 <span className="uir-text-field-left-icon">
-                    {this.props.icon}
+                    <Button
+                        className="icon-wrapper"
+                        icon={this.props.icon}
+                        onClick={this.handleClearIconClick}
```
Comment (jfrej) [inline src/components/Fields/TextField.js]: This shouldn't be here, right? Now when I click on any icon, the field clears.

File: src/components/Fields/TextField.js [hunk 2]
```diff
@@ -265,19 +278,24 @@ class TextField extends Component {
...
+                        variant={ButtonVariant.CLEAR}
+                    />
                 </span>
             ) :
             null;
         const clearIcon = this.props.isClearable && this.state.value ?
             (
                 <span className="uir-text-field-right-icon">
                     <Button
+                        className="icon-wrapper"
```
Comment (jfrej) [inline src/components/Fields/TextField.js]: Should this class be prefixed with something?

File: src/components/Fields/TextField.scss [hunk 3]
```diff
@@ -128,44 +128,36 @@ $textFieldLabelColor: $charcoal !default;
...
+    .uir-button.icon-wrapper {
+        min-width: 16px;
+        padding: 0;
+        margin: 0;
 
-            &:hover {
-                background-color: transparent;
-            }
+        &:hover {
+            background-color: transparent;
```
Comment (jfrej) [inline src/components/Fields/TextField.scss]: This should also be added to `:focus` - currently there's a grey background on focus.

File: src/components/Fields/TextField.js (hunk 1 above)
Comment (jfrej) [reply src/components/Fields/TextField.js]: Should we have a test that checks that the field doesn't clear ;) ?

File: src/components/Fields/TextField.scss (hunk 3 above)
Comment (mattpauldavies) [reply src/components/Fields/TextField.scss]: ahh thanks well spotted

File: src/components/Fields/TextField.js (hunk 2 above)
Comment (mattpauldavies) [reply src/components/Fields/TextField.js]: possibly... I initially thought `uir-text-field-icon-wrapper` was a bit long!! but perhaps that's the best way to go

File: src/components/Fields/TextField.scss [hunk 4]
```diff
@@ -128,44 +128,37 @@ $textFieldLabelColor: $charcoal !default;
...
+        &:focus,
+        &:hover {
+            background-color: transparent;
+        }
 
-            .uir-icon-clear:hover {
-                fill: $charcoal;
-            }
+        .uir-icon-clear:hover {
+            fill: $charcoal;
```
Comment (jfrej) [inline src/components/Fields/TextField.scss]: It'd be nice to add `:focus` to this as well. Currently there is no indication of focus when I tab into the clear icon.

**Pull request merged**